### PR TITLE
feat(release): use action git auth

### PR DESCRIPTION
**Pull request by tagoro9**
**Description**

So that commands down the line that use git are correctly authenticated.

**Changes**

* feat(release): use action git auth

🚀 PR created with [fotingo](https://github.com/tagoro9/fotingo)

### CODE REVIEW

File: release/action.yaml [hunk 1]
```diff
@@ -34,10 +34,13 @@ runs:
   steps:
     - name: Checkout
       uses: actions/checkout@v3
-      if: inputs.checkout-repo
```
Comment (shakefu) [inline release/action.yaml:37]: Is this a mistaken remove? If not the input should be pulled

**Review by shakefu — CHANGES_REQUESTED** [review]

File: release/action.yaml (hunk 1 above)
Comment (tagoro9) [reply release/action.yaml:37]: This removed on purpose. See how it is used in the line below. It was a bug from a previous pr

Comment (dgschwindturo) [inline release/action.yaml:37]: Why is this guard clause being removed? Looks like this should instead be `if: inputs.checkout-repo == 'true'` and that particular guard clause is not consistent throughout `open-turo`, which I will address.

**Review by dgschwindturo — COMMENTED** [review]
Wondering why that guard clause would be removed?

File: release/action.yaml (hunk 1 above)
Comment (tagoro9) [reply release/action.yaml:37]: This is removed because it's now used in the line before. I'm on m phone but see the latest merged pr for more context

Comment (tagoro9) [reply release/action.yaml:37]: This guard is just meant to be used for the force depth setting in the line below

Comment (tagoro9) [reply release/action.yaml:37]: Now I see the error. Undid this change and my other stupid change from the previous PR

**Pull request closed without merging**